design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,278 @@
+type Customer {
+    gitft_registry_list: [GiftRegistry]
+    gitft_registry(id: ID!): GiftRegistry
+}
+
+type Query {
+    giftRegistryTypes: [GiftRegistryType] @doc(description: "Get a list of available gift registry types")
+    giftRegistrySearch(
+        registrant_firstname: String!,
+        registrant_lastname: String!,
+        gift_registry_type_id: ID,
+        searchable_dynamic_attributes: [GiftRegistryDynamicAttributeInput] @doc(description: "For select attributes ID should be provided expected. For range search, '_from' and '_to' suffixes can be added to the attribute code. For text attributes provide value for exact matching.")
+    ): [GiftRegistry] @doc(description: "Gift registry search by registrant name and additional searchable attributes.")
+}
+
+type Mutation {
+    # All mutations below should only be accessible to the registry owner
+    createGiftRegistry(gift_registry: CreateGiftRegistryInput): CreateGiftRegistryOutput
+    updateGiftRegistry(id: ID!, gift_registry: UpdateGiftRegistryInput): UpdateGiftRegistryOutput
+    addProductsToGiftRegistry(gift_registry_id: ID!, items: [GiftRegistryItemInput!]!): AddProductsToGiftRegistryOutput
+}
+
+input GiftRegistryItemInput {
+    sku: String!
```

Comment:
Any reason for `sku` and `parent_sku` to return a `String` and not the actual concrete entity?

Reply:
This is input type, not output. In the output the actual product entity is returned.

Reply:
Thanks for pointing that out!